Natoms and nreplicas

GiovanniBussi opened the pull request:
@maxbonomi @carlocamilloni I would like to merge this. I implemented the possibility to control the number of replicas (default is 0, that implies not to use mpiexec).

I changed a bit Carlo's implementation to control number of atoms (number of replicas is done in the same way). Now you can do:
````
url: ...
natoms: 100
````
to change all input files
````
url: ...
plumed_input:
  - first.dat
  - path: second.dat
    natoms: 100
````
to change only the number of atoms for the second file. In practice, for each input file, if it is a string I interpret it as the path, if it is a dictionary I allow one to provide separately path, natoms, and nreplicas.

I think I also fixed the eggs that needed special treatments (005 and 010)

maxbonomi commented:
> On Apr 22, 2019, at 13:20, Giovanni <notifications@github.com> wrote:
> 
> @maxbonomi @carlocamilloni I would like to merge this. I implemented the possibility to control the number of replicas (default is 0, that implies not to use mpiexec).
> 
> I changed a bit Carlo's implementation to control number of atoms (number of replicas is done in the same way). Now you can do:
> 
> url: ...
> natoms: 100
> 
> to change all input files
> 
> url: ...
> plumed_input:
>   - first.dat
>   - path: second.dat
>     natoms: 100

Would it be hard to also for the first case to specify
path: first.dat

?
Just to have a uniform notation, with some compulsory keywords (only path?) and other optional

Max




> to change only the number of atoms for the second file. In practice, for each input file, if it is a string I interpret it as the path, if it is a dictionary I allow one to provide separately path, natoms, and nreplicas.
> 
> I think I also fixed the eggs that needed special treatments (005 and 010)
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/plumed-nest/plumed-nest/pull/35
> 
> Commit Summary
> 
> 	• test
> 	• control nreplicas
> File Changes
> 
> 	• M eggs-19/005/nest.yml (1)
> 	• M eggs-19/010/nest.yml (6)
> 	• M eggs-19/012/nest.yml (9)
> 	• M nest.py (43)
> Patch Links:
> 
> 	• https://github.com/plumed-nest/plumed-nest/pull/35.patch
> 	• https://github.com/plumed-nest/plumed-nest/pull/35.diff
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

GiovanniBussi commented:
I think it should work already also if you use path:. The reason I allowed
this is that I thought in most cases one wants to just specify the list of
flies. If more info are needed, them you use a dictionary with path and
other keys.

What do you think?


Il giorno lun 22 apr 2019 alle 13:28 Massimiliano Bonomi <
notifications@github.com> ha scritto:

>
>
> > On Apr 22, 2019, at 13:20, Giovanni <notifications@github.com> wrote:
> >
> > @maxbonomi @carlocamilloni I would like to merge this. I implemented the
> possibility to control the number of replicas (default is 0, that implies
> not to use mpiexec).
> >
> > I changed a bit Carlo's implementation to control number of atoms
> (number of replicas is done in the same way). Now you can do:
> >
> > url: ...
> > natoms: 100
> >
> > to change all input files
> >
> > url: ...
> > plumed_input:
> > - first.dat
> > - path: second.dat
> > natoms: 100
>
> Would it be hard to also for the first case to specify
> path: first.dat
>
> ?
> Just to have a uniform notation, with some compulsory keywords (only
> path?) and other optional
>
> Max
>
>
>
>
> > to change only the number of atoms for the second file. In practice, for
> each input file, if it is a string I interpret it as the path, if it is a
> dictionary I allow one to provide separately path, natoms, and nreplicas.
> >
> > I think I also fixed the eggs that needed special treatments (005 and
> 010)
> >
> > You can view, comment on, or merge this pull request online at:
> >
> > https://github.com/plumed-nest/plumed-nest/pull/35
> >
> > Commit Summary
> >
> > • test
> > • control nreplicas
> > File Changes
> >
> > • M eggs-19/005/nest.yml (1)
> > • M eggs-19/010/nest.yml (6)
> > • M eggs-19/012/nest.yml (9)
> > • M nest.py (43)
> > Patch Links:
> >
> > • https://github.com/plumed-nest/plumed-nest/pull/35.patch
> > • https://github.com/plumed-nest/plumed-nest/pull/35.diff
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub, or mute the thread.
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/plumed-nest/plumed-nest/pull/35#issuecomment-485396589>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABBNCXUF42D7E3ZHK7OODILPRWONBANCNFSM4HHPDSPA>
> .
>
-- 
Sent from Gmail mobile

maxbonomi commented:
You are right, cases in which you need to specify natoms are not that common.
Ok for merging for me!


> On Apr 22, 2019, at 13:38, Giovanni <notifications@github.com> wrote:
> 
> I think it should work already also if you use path:. The reason I allowed
> this is that I thought in most cases one wants to just specify the list of
> flies. If more info are needed, them you use a dictionary with path and
> other keys.
> 
> What do you think?
> 
> 
> Il giorno lun 22 apr 2019 alle 13:28 Massimiliano Bonomi <
> notifications@github.com> ha scritto:
> 
> >
> >
> > > On Apr 22, 2019, at 13:20, Giovanni <notifications@github.com> wrote:
> > >
> > > @maxbonomi @carlocamilloni I would like to merge this. I implemented the
> > possibility to control the number of replicas (default is 0, that implies
> > not to use mpiexec).
> > >
> > > I changed a bit Carlo's implementation to control number of atoms
> > (number of replicas is done in the same way). Now you can do:
> > >
> > > url: ...
> > > natoms: 100
> > >
> > > to change all input files
> > >
> > > url: ...
> > > plumed_input:
> > > - first.dat
> > > - path: second.dat
> > > natoms: 100
> >
> > Would it be hard to also for the first case to specify
> > path: first.dat
> >
> > ?
> > Just to have a uniform notation, with some compulsory keywords (only
> > path?) and other optional
> >
> > Max
> >
> >
> >
> >
> > > to change only the number of atoms for the second file. In practice, for
> > each input file, if it is a string I interpret it as the path, if it is a
> > dictionary I allow one to provide separately path, natoms, and nreplicas.
> > >
> > > I think I also fixed the eggs that needed special treatments (005 and
> > 010)
> > >
> > > You can view, comment on, or merge this pull request online at:
> > >
> > > https://github.com/plumed-nest/plumed-nest/pull/35
> > >
> > > Commit Summary
> > >
> > > • test
> > > • control nreplicas
> > > File Changes
> > >
> > > • M eggs-19/005/nest.yml (1)
> > > • M eggs-19/010/nest.yml (6)
> > > • M eggs-19/012/nest.yml (9)
> > > • M nest.py (43)
> > > Patch Links:
> > >
> > > • https://github.com/plumed-nest/plumed-nest/pull/35.patch
> > > • https://github.com/plumed-nest/plumed-nest/pull/35.diff
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub, or mute the thread.
> > >
> >
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/plumed-nest/plumed-nest/pull/35#issuecomment-485396589>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/ABBNCXUF42D7E3ZHK7OODILPRWONBANCNFSM4HHPDSPA>
> > .
> >
> -- 
> Sent from Gmail mobile
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
